### PR TITLE
add: gcc bpf cross compiler

### LINKED ISSUE #4408: [COMPILER REQUEST]: LLVM and GCC BPF
### Compiler name

GCC and LLVM

### Compiler version

latest 

### Compiler language

all of them :)

### Compiler homepage

-

### Motivation

As discussed with , BPF support in CE would be nice.
- [x] first step is having BPF supported in crosstool-ng https://github.com/crosstool-ng/crosstool-ng/pull/1870
- [x] then setup our builders to build an x-gcc for bpf
- [x] update compilers accordingly (what languages?) -> C/C++
- [x] see if monaco has any support for syntax highlighting BPF (-> there's none)

### PR DESCRIPTION
**Pull request by dkm**
Add GCC for C and C++, for BPF target.

fixes #4408

Signed-off-by: Marc Poulhiès <dkm@kataplop.net>

### CODE REVIEW

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -861,14 +861,32 @@ compiler.zapcc190308.name=x86-64 Zapcc 190308
...
+group.gccbpf.supportsExecute=false
+group.gccbpf.baseName=BPF gcc
+group.gccbpf.groupName=BPF GCC
+group.gccbpf.isSemVer=true
+
+compiler.bpfg1220.exe=/opt/compiler-explorer/bpf/gcc-12.2.0/bpf-unknown-none/bin/bpf-unknown-none-g++
+compiler.bpfg1220.semver=12.2.0
+compiler.bpfg1220.objdumper=/opt/compiler-explorer/bpf/gcc-12.2.0/bpf-unknown-none/bin/bpf-unknown-objdump
+compiler.bpfg1220.demangler=/opt/compiler-explorer/bpf/gcc-12.2.0/bpf-unknown-none/bin/bpf-unknown-none-c++filt
+compiler.bpfg1220.name=bpf 12.2.0
```
Comment (RubenRBS) [inline etc/config/c++.amazon.properties:888]: No need to be explicit here for compilers that both define `baseName` and set `isSemver: true`, tne `.name` will equal `<baseName> <semVer>`

**Review by RubenRBS — APPROVED** [review]

File: etc/config/c++.amazon.properties (hunk 1 above)
Comment (dkm) [reply etc/config/c++.amazon.properties:888]: ahhhh sorry, I forgot to update my script that generates these lines

Comment (dkm) [reply etc/config/c++.amazon.properties]: https://github.com/compiler-explorer/gcc-cross-builder/pull/46 <= that should be the last time you need to remind me to remove this line /o\

Comment (RubenRBS) [reply etc/config/c++.amazon.properties]: :)))

**Comment by dkm** [comment]
Hopefully, first automated nightly build tonight

**Review by RubenRBS — APPROVED** [review]

**Comment by jeremy-rifkin** [comment]
This is now live

**Pull request merged**